onadata/apps/logger/models/xform.py

```diff
@@ -1415,6 +1415,28 @@ def save_project(sender, instance=None, created=False, **kwargs):
 pre_save.connect(save_project, sender=XForm, dispatch_uid="save_project_xform")
 
 
+# pylint: disable=unused-argument,import-outside-toplevel
+def _create_meta_perms(sender, instance, created, **kwargs):
+    metadata = instance.metadata_set.all().filter(data_type="xform_meta_perms").first()
+    if instance.deleted_by is None and metadata is None:
+        from onadata.libs.serializers.metadata_serializer import MetaDataSerializer
+
+        serializer = MetaDataSerializer(
```

Comment:
In the past, how have created `xform_meta_perms`? It's best to use the model here to created the object directly instead of using the serializer

Reply:
the `xform_meta_perms` , in the past, has been created by making API requests from the front-end. So this record is normally not created on the back-end service

Reply:
We could use the `MetaData.objects.create` method directly. I think if we use the serializer this way, we will benefit from some validation and logic already in the `create` method of the serializer

Reply:
@FrankApiyo Noted, though since it's being created directly and not a request from a client, we are assured the values are always correct. But if there is some essential checks that we think applies, it should be moved to a some utiility module accessible by the serializer module and model module

Reply:
I'll move this into a different PR